### PR TITLE
DOCS: Fixing broken links in documentation.

### PR DESCRIPTION
**Pull request by sgolebiewski-intel**
This fixes the broken links found throughout the OV docs.

### CODE REVIEW

File: thirdparty/cnpy/README.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ Loading data written in numpy formats into C++ is equally simple, but requires y
 Default installation directory is /usr/local. 
 To specify a different directory, add `-DCMAKE_INSTALL_PREFIX=/path/to/install/dir` to the cmake invocation in step 4.
 
-1. get [cmake](www.cmake.org)
+1. get [cmake](https://cmake.org/)
```
Comment (ilya-lavrenov) [inline thirdparty/cnpy/README.md:19]: I suppose we need to contribute it to https://github.com/rogersce/cnpy

Comment (sgolebiewski-intel) [reply thirdparty/cnpy/README.md:19]: What do you propose?

Comment (ilya-lavrenov) [reply thirdparty/cnpy/README.md:19]: send this change to that repo as well.

Comment (msmykx-intel) [reply thirdparty/cnpy/README.md:19]: I contributed this fix for https://github.com/rogersce/cnpy

Comment (sgolebiewski-intel) [reply thirdparty/cnpy/README.md:19]: I guess we can resolve this conversation :)

File: docs/optimization_guide/model_optimization_guide.md [hunk 2]
```diff
@@ -17,11 +17,11 @@
...
+- :doc:`Model Optimizer <openvino_docs_MO_DG_Deep_Learning_Model_Optimizer_DevGuide>` implements most of the optimization parameters to a model by default. Yet, you are free to configure mean/scale values, batch size, RGB vs BGR input channels, and other parameters to speed up preprocess of a model (:doc:`Embedding Preprocessing Computation <openvino_docs_MO_DG_Additional_Optimization_Use_Cases>`).
 
-- :ref:`Post-training Quantization` is designed to optimize inference of deep learning models by applying post-training methods that do not require model retraining or fine-tuning, for example, post-training 8-bit integer quantization.
+- :ref:`Post-training Quantization <pot_introduction>` is designed to optimize inference of deep learning models by applying post-training methods that do not require model retraining or fine-tuning, for example, post-training 8-bit integer quantization.
```
Comment (msmykx-intel) [inline docs/optimization_guide/model_optimization_guide.md:22]:
```suggestion
- :doc:`Post-training Quantization <pot_introduction>` is designed to optimize inference of deep learning models by applying post-training methods that do not require model retraining or fine-tuning, for example, post-training 8-bit integer quantization.
```

File: docs/optimization_guide/model_optimization_guide.md [hunk 3]
```diff
@@ -17,11 +17,11 @@
...
 
-- :ref:`Training-time Optimization`, a suite of advanced methods for training-time model optimization within the DL framework, such as PyTorch and TensorFlow 2.x. It supports methods, like Quantization-aware Training and Filter Pruning. NNCF-optimized models can be inferred with OpenVINO using all the available workflows.
+- :ref:`Training-time Optimization <nncf_ptq_introduction>`, a suite of advanced methods for training-time model optimization within the DL framework, such as PyTorch and TensorFlow 2.x. It supports methods, like Quantization-aware Training and Filter Pruning. NNCF-optimized models can be inferred with OpenVINO using all the available workflows.
```
Comment (msmykx-intel) [inline docs/optimization_guide/model_optimization_guide.md:24]:
```suggestion
- :doc:`Training-time Optimization <nncf_ptq_introduction>`, a suite of advanced methods for training-time model optimization within the DL framework, such as PyTorch and TensorFlow 2.x. It supports methods, like Quantization-aware Training and Filter Pruning. NNCF-optimized models can be inferred with OpenVINO using all the available workflows.
```

**Review by kblaszczak-intel — APPROVED** [review]

**Pull request merged**